cmd/govim/complete.go

```diff
@@ -61,6 +61,35 @@ func (v *vimstate) complete(args ...json.RawMessage) (interface{}, error) {
 	}
 }
 
+func (v *vimstate) completeDone(args ...json.RawMessage) error {
+	currBufNr := v.ParseInt(args[0])
+	b, ok := v.buffers[currBufNr]
+	if !ok {
+		return fmt.Errorf("failed to resolve buffer %v", currBufNr)
+	}
+	var chosen struct {
+		Word string `json:"word"`
+		Abbr string `json:"abbr"`
+	}
+	v.Parse(args[1], &chosen)
+	if chosen.Word == "" {
+		return nil
+	}
+	var match *protocol.CompletionItem
+	for _, c := range v.lastCompleteResults.Items {
+		if c.Label == chosen.Abbr {
+			match = &c
+		}
+	}
+	if match == nil {
+		return fmt.Errorf("failed to find match for completed item %#v", chosen)
```

Comment:
This might actually happen in a OK state, if someone performs a content-based completion (after, at some point, having performed a omnicomplete to prevent the panic above).

Reply:
Thanks, per our Slack chat I've added logic per Bram's response here https://groups.google.com/d/msg/vim_dev/z7acHIqGMBE/GoGpGnGzAwAJ